src/main/java/seedu/address/logic/commands/todolistcommands/EditTaskCommand.java

```diff
@@ -92,10 +94,10 @@ private static Task createEditedTask(Task taskToEdit, EditTaskDescriptor editTas
 
         Task editedTask = new Task(name);
 
-        if (taskToEdit.getTag().isPresent() || editTaskDescriptor.getTag().isPresent()) {
-            Tag updatedTag = editTaskDescriptor.getTag()
-                    .orElse(taskToEdit.getTag().get());
-            editedTask = editedTask.setTag(updatedTag);
+        if (taskToEdit.getTags().isPresent() || editTaskDescriptor.getTags().isPresent()) {
+            Set<Tag> updatedTag = editTaskDescriptor.getTags()
+                    .orElse(taskToEdit.getTags().get());
+            editedTask = editedTask.setTags(updatedTag);
         }
 
         if (taskToEdit.getPriority().isPresent() || editTaskDescriptor.getPriority().isPresent()) {
```

Comment:
The same bug as the one mentioned earlier might occur within this if block.

Reply:
Okay, noted.